src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/CreateTake.kt

```diff
@@ -0,0 +1,134 @@
+        val rcSlug = project.resourceContainer?.identifier ?: ""
+        val bookNumber = "%02d".format(
+                // Handle book number offset (only for Bibles)
+                if (project.resourceContainer?.subject?.toLowerCase() == "bible"
+                        && project.sort > 39) project.sort + 1 else project.sort
+        )
+        val bookSlug = project.slug
+        val chapterNumber = chapterFormat.format(chapter.titleKey.toInt())
+        val verseNumber = if (chunk.start == chunk.end) {
+            verseFormat.format(chunk.start)
+        } else {
+            "$verseFormat-$verseFormat".format(chunk.start, chunk.end)
+        }
+        val takeNumber = "%02d".format(number)
+
+        // Compile the complete filename
+        return listOf(
+                languageSlug,
+                rcSlug,
+                "b$bookNumber",
+                bookSlug,
+                "c$chapterNumber",
+                "v$verseNumber",
+                "t$takeNumber"
+        ).joinToString("_", postfix = ".wav")
+    }
+
+    fun create(chunk: Chunk, project: Collection, chapter: Collection): Single<Take> = Single
+            .zip(
+                    getHighestTakeNumber(chunk),
+                    getNumberOfSubcollections(project),
+                    getNumberOfChunks(chapter),
+                    Function3 { highest, chapterCount, verseCount ->
+                        val filename = generateTakeFilename(
+                                project,
+                                chapter,
+                                chunk,
+                                highest + 1,
+                                chapterCount,
+                                verseCount
+                        )
+
+                        // Create a file for this take
+                        val chapterFormat = if (chapterCount > 99) "%03d" else "%02d"
+
+                        val takeFile = directoryProvider
+                                .getProjectAudioDirectory(project, chapterFormat.format(chapter.titleKey.toInt()))
+                                .resolve(File(filename))
+
+                        val newTake = Take(
+                                takeFile.name,
+                                takeFile,
+                                highest + 1,
+                                LocalDate.now(),
+                                false,
+                                listOf() // No markers
+                        )
+
+                        // Create an empty WAV file
+                        waveFileCreator.createEmpty(newTake.path)
+                        return@Function3 newTake
+                    }
+            )
+
+    fun saveNew(take: Take, chunk: Chunk): Completable = takeRepository
+            .insertForChunk(take, chunk)
+            .toCompletable()
+
+    fun recordAndSaveNewTake(chunk: Chunk, project: Collection, chapter: Collection): Completable {
```

Comment:
the and is a pretty good indicator that this function is doing too much;

Focus on creating the file, let some kind of audio recorder write to it

Reply:
Save here means inserting into the database. This function is not writing to the audio file directly. Instead uses the create and saveNew functions along with launching the plugin to complete the task

Reply:
If naming should be changed, it could become just `record`